### ISSUE TITLE
Promote powerpc64-unknown-freebsd to Tier 2

### ISSUE THREAD

**Issue by pkubaj** [rust-lang/rust#80586]
We are interested in making powerpc64 (and in the future powerpc64le) Tier 2 in Rust, to make it easier to test nightly builds in our package cluster. What would be necessary to do it?

Some other powerpc* targets (for Linux) are already Tier 2.

**Comment by pkubaj** [comment]
Our ELFv1 port currently requires external patch to link with non-base compiler (that is non-upstreamable).

However, ELFv2 currently works without any patches (apart from replacing ELFv1 with ELFv2 in `compiler/rustc_target/src/abi/call/powerpc64.rs` and adding ELFv2 to `compiler/rustc_target/src/spec/powerpc64_unknown_freebsd.rs`). Once https://github.com/rust-lang/rfcs/pull/2992  lands, we'll be able to just set ELFv2 in our platform config, so there will be no patching.

Since https://github.com/rust-lang/rfcs/pull/2992 seems to be ready, I expect this issue will soon be fixed. It will be enough if only ELFv2 gets proper support, ELFv1 is long obsolete and we can just patch it on our own until we decide to drop it.

**Comment by lzutao** [comment]
cc @pietroalbini or @Mark-Simulacrum as infra team members.

**Comment by Mark-Simulacrum** [comment]
I am inclined to wait for https://github.com/rust-lang/rfcs/pull/2803 to get approved before we examine new targets or tier movement for existing targets.

**Comment by sanxiyn** [comment]
Now target tier policy RFC is merged, can we proceed here?

**Comment by pkubaj** [comment]
I'm quite unsure how to solve the ABI issue. FreeBSD targets ELFv2 ABI on big-endian, same as e.g. OpenBSD, Void Linux and Adelie Linux (so we're not alone).
We currently use those patches:
https://cgit.freebsd.org/ports/tree/lang/rust/files/powerpc64-elfv2/patch-compiler_rustc__target_src_abi_call_powerpc64.rs
https://cgit.freebsd.org/ports/tree/lang/rust/files/powerpc64-elfv2/patch-compiler_rustc__target_src_spec_powerpc64__unknown__freebsd.rs

FreeBSD will support big-endian POWER and there are no plans to drop it, but given that there's no way to specify ABI and we need to patch Rust, I think it would be nice to switch to PR to refer powerpc64le-unknown-freebsd target.
In POWER world, little-endian is generally considered the future and, while the newer CPUs support big-endian as well, most people nowadays run little-endian.

Is that reasonable?

**Comment by workingjubilee** [comment]
Is https://github.com/rust-lang/rust/issues/80970 inappropriate for specifying/conditionalizing on ABI?

**Comment by pkubaj** [comment]
It's not really necessary. powerpc64le-unknown-freebsd supports only ELFv2, so it will be appriopriate to just hardcode the ABI. On the other hand, powerpc64-unknown-freebsd supports both ELFv1 and ELFv2, so specifying ABI would be helpful, but ELFv1 is just FreeBSD 12.4-RELEASE (of the supported versions) and that one will be supported just until the end of year. We removed support for Rust on ELFv1 in FreeBSD ports anyway and the users are advised to migrate to at least 13.x (which is ELFv2). I believe it will be fine to just hardcode ELFv2 as well at this point.

**Comment by workingjubilee** [comment]
Oh. In that case, because these are tier 3 targets currently and thus much more subject to change, and because the goal would be to demand ELFv2 anyways, I believe that it would be more appropriate to simply change things as-necessary to enforce that they use ELFv2 _before_ elevating anything to tier 2.

It's fine if, in doing this, two FreeBSD targets would demand different FreeBSD OS version minimums, at your preference. We don't demand that sort of consistency.

**Comment by pkubaj** [comment]
What would be required to fix this issue? I just hit another issue, which requires Rust to be at least Tier 2 to be fixed: https://github.com/rollup/rollup/pull/6398
